flink-autoscaler/src/main/java/org/apache/flink/autoscaler/JobAutoScalerImpl.java

```diff
@@ -106,7 +107,13 @@ public void scale(Context ctx) throws Exception {
         } catch (Throwable e) {
             onError(ctx, autoscalerMetrics, e);
         } finally {
-            applyParallelismOverrides(ctx);
+            try {
+                applyParallelismOverrides(ctx);
+                applyConfigOverrides(ctx);
```

Comment:
This seems to be causing some issues when both parallelism and memory configurations get changed and in-place scaling is enabled. I could not figure out why, but I'll try to look into it next week.

I see the following error coming from the reconciliation loop:
```
Could not create Kubernetes cluster "test-app-scaling"
...
Message: object is being deleted: deployments.apps \"test-app-scaling\" already exists.
```

Reply:
Oh right, good point. In-place scaling won't be possible with auto tuning because the Flink config can't be updated on the fly. We need to fall back to a full upgrade in this case.

Reply:
If we apply the config overrides on the FlinkDeployment spec itself then this will be done automatically.

Reply:
Actually, this is already handled by the current code. The in-place parallelism class doesn't apply config overrides. Need to look into this further.

Reply:
Sorry, it's not related to this change at all, and it's completely intermittent. I seem to be running into FLINK-32334 and the TaskManagers (thus the Kubernetes Deployment as well) are not properly deleted in native mode before starting the new deployment.